### PR TITLE
Added Serverless overview

### PR DESCRIPTION
**Pull request by abrennan89**
Added Serverless overview assembly and architecture module.

### CODE REVIEW

**Comment by abrennan89** [comment]
branch/enterprise-4.1

File: modules/serverless-architecture.adoc [hunk 1]
```diff
@@ -0,0 +1,35 @@
+// Module included in the following assemblies:
+
+// * serverless-overview.adoc
+
+[id="serverless-architecture_{context}"]
+= Red Hat OpenShift Serverless product architecture
+
+toc::[]
```
Comment (kalexand-rh) [inline modules/serverless-architecture.adoc]: You don't need this line in modules.

File: modules/serverless-architecture.adoc [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+==== Knative configuration resource
+The `configuration.serving.knative.dev` resource maintains the desired state for your deployment. Modifying a configuration creates a new revision.
+
+==== Knative revision resource
+The `revision.serving.knative.dev` resource is a point-in-time snapshot of the code and configuration for each modification made to the workload. Revisions are immutable objects and can be retained for as long as needed.
+Cluster administrators can modify the `revision.serving.knative.dev` resource to enable automatic scaling of pods in your OpenShift cluster.
+
+== Knative Client
+Knative Client (`kn`) extends the functionality of `oc` or `kubectl` to enable interaction with Knative components on OpenShift. `kn` allows developers to deploy and manage applications without editing YAML files directly.
```
Comment (kalexand-rh) [inline modules/serverless-architecture.adoc]: The Knative Client

I'd s/of `oc` or `kubectl`/ of the `oc` or `kubectl` tools  (https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/term_glossary.adoc#openshift-cli)

File: serverless/serverless-overview.adoc [hunk 3]
```diff
@@ -0,0 +1,27 @@
+[id="serverless-overview"]
+= Red Hat OpenShift Serverless overview
+include::modules/common-attributes.adoc[]
+:context: serverless-overview
+
+toc::[]
+
+[IMPORTANT]
```
Comment (kalexand-rh) [inline serverless/serverless-overview.adoc]: It's similar to the language in the [tech preview module](https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#indicating-technology-preview-features). Where did you source this note? I'm curious if the tech preview module's fallen out of date.

File: serverless/serverless-overview.adoc [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+Red Hat OpenShift Serverless is a Technology Preview feature only. Technology Preview features are
+not supported with Red Hat production service level agreements (SLAs) and might not be
+functionally complete. Red Hat does not recommend using them in production.
+These features provide early access to upcoming product features, enabling
+customers to test functionality and provide feedback during the development process.
+For more information about the support scope of Red Hat Technology Preview features,
+see link:https://access.redhat.com/support/offerings/techpreview/[Red Hat Technology Preview Features Support Scope].
+====
+
+Red Hat OpenShift Serverless is based on the open source Knative project, which provides portability and consistency across hybrid and multi-cloud environments by enabling an enterprise-grade serverless platform. It implements the building blocks for developers to create modern, source-centric, container-based applcations through a series of CRDs (Custom Resource Definition) and associated controllers in Kubernetes.
```
Comment (kalexand-rh) [inline serverless/serverless-overview.adoc]: For "CRDs (Custom Resource Definition)," flip the order and use (CRDs)

File: serverless/serverless-overview.adoc [hunk 5]
```diff
@@ -0,0 +1,27 @@
+[id="serverless-overview"]
+= Red Hat OpenShift Serverless overview
```
Comment (kalexand-rh) [inline serverless/serverless-overview.adoc]: Is this the overview or the architecture guide?

Are you planning to add more modules to this assembly? If it's complete, I think that you need to retitle the assembly to say that it's the arch guide and adjust the module's initial headings.

**Review by kalexand-rh — COMMENTED** [review]
I have some suggestions and questions, but I think the content's good.

File: _topic_map.yml [hunk 6]
```diff
@@ -933,3 +933,49 @@ Topics:
...
     File: cnv-release-notes
+#---
+#Name: Serverless applications
+#Dir: serverless
+#Distros: openshift-enterprise
+#Topics:
+#- Name: Serverless overview
+#  File: serverless-overview
+#---
+#Name: Container-native Virtualization
```
Comment (kalexand-rh) [inline _topic_map.yml]: Will you check your TOC to make sure that you're just changing the serverless section? I'm not sure how you picked up service mesh and CNV changes, but we don't want to drop this content from the TOC.

File: serverless/serverless-overview.adoc (hunk 3 above)
Comment (abrennan89) [reply serverless/serverless-overview.adoc]: Fixed, pinged you on Slack re the source.

File: _topic_map.yml (hunk 6 above)
Comment (abrennan89) [reply _topic_map.yml]: There were two instances in my file, maybe caused by a rebase or something.
This one at the bottom seems to have been a placeholder, I checked which is the right one from the current repo and removed the 'placeholder' versions.

File: serverless/serverless-overview.adoc (hunk 5 above)
Comment (abrennan89) [reply serverless/serverless-overview.adoc]: Separated these - I was inc. the architecture in this overview but it's out now.

File: modules/serverless-architecture.adoc [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+== Knative Serving
+
+Knative Serving on OpenShift builds on Kubernetes and Maistra to support deploying and serving serverless applications.
+
+It creates a set of Kubernetes Custom Resource Definitions (CRDs) that are used to define and control the behavior of serverless workloads on an OpenShift cluster.
+
+These CRDs can be used as building blocks to address complex use cases, such as rapid deployment of serverless containers, automatic scaling of pods, routing and network programming for Maistra components, or viewing point-in-time snapshots of deployed code and configurations.
+
+=== Knative Serving components
```
[the remark this answers is not in the document]
Comment (abrennan89) [reply modules/serverless-architecture.adoc]: Added a line and fixed the list

File: serverless/serverless-architecture.adoc [hunk 8]
```diff
@@ -0,0 +1,33 @@
+// Module included in the following assemblies:
+
+// * serverless-overview.adoc
+
+[id="serverless-architecture_{context}"]
```
Comment (vikram-redhat) [inline serverless/serverless-architecture.adoc]: @abrennan89 assemblies don't have context as an attribute to their anchor ids. See the assembly file structure: https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#assembly-file-metadata

Comment (abrennan89) [reply serverless/serverless-architecture.adoc]: Neither of these should be assemblies now, think they're just in the wrong place. I moved them into modules.

Comment (abrennan89) [reply serverless/serverless-architecture.adoc]: Sorry, the overview one is I guess, it includes the TP module.

**Comment by abrennan89** [comment]
@vikram-redhat @kalexand-rh files should be fixed and all the changes are in now. LMK if you have any other feedback, and thanks for the help! :)

**Comment by abrennan89** [comment]
> @vikram-redhat @kalexand-rh files should be fixed and all the changes are in now. LMK if you have any other feedback, and thanks for the help! :)

Scratch that, I'm finally getting to grips with all the OpenShift docs docs :joy: so I will review this in terms of the mod docs templates one more time first. I'll post in Slack when this is ready for a second review. Thanks so much for your help.

**Comment by kalexand-rh** [comment]
@vikram-redhat, I'm adding this to 4.2 as well. Please revert if you want this change in just 4.1.

File: serverless/understanding-serverless.adoc [hunk 9]
```diff
@@ -0,0 +1,14 @@
+[id="understanding-serverless"]
+= Understanding Red Hat OpenShift Serverless
+include::modules/common-attributes.adoc[]
+:context: understanding-serverless
+
+:FeatureName: Red Hat OpenShift Serverless
+include::modules/technology-preview.adoc[leveloffset=+1]
+
+Red Hat OpenShift Serverless is based on the open source Knative project, which provides portability and consistency across hybrid and multi-cloud environments by enabling an enterprise-grade serverless platform. It implements the building blocks for developers to create modern, source-centric, container-based applcations through a series of Custom Resource Definitions (CRDs) and associated controllers in Kubernetes.
```
Comment (abrennan89) [inline serverless/understanding-serverless.adoc]: @abrennan89 reminder; typo in 'applications'

File: modules/knative-serving.adoc [hunk 10]
```diff
@@ -0,0 +1,26 @@
+// Module included in the following assemblies:
+//
+// * serverless/serverless-architecture.adoc
+
+[id="knative-serving_{context}"]
+= Knative Serving
+
+Knative Serving on {product-title} builds on Kubernetes and Maistra to support deploying and serving serverless applications.
```
Comment (abrennan89) [inline modules/knative-serving.adoc]: change Maistra to Istio in all instances

**Review by abrennan89 — COMMENTED** [review]
@abrennan89 reminder to fix these minor issues when un-commenting the topic map yml entry

**Pull request merged**